src/outbox/outboxActions.js

```diff
@@ -0,0 +1,22 @@
+/* @flow */
+import type { Dispatch, GetState } from '../types';
+import { MESSAGE_SEND } from '../actionConstants';
+import { getAuth } from '../selectors';
+import { attemptToSendMessages } from './outboxMessageActions';
+
+export const sendMessageAction = (params: Object) => ({
+  type: MESSAGE_SEND,
+  params,
+});
+
+export const addToOutbox = (
+  type: 'private' | 'stream',
+  to: string | string[],
+  subject: string,
+  content: string,
+) => async (dispatch: Dispatch, getState: GetState) => {
+  dispatch(sendMessageAction({ type, to, subject, content, localMessageId: new Date().getTime() }));
```

Comment:
@timabbott I've used the `new Date().getTime()` as the local id and avoided using indices as I am using an array here! 
This should be fine, right?